Example request to update the retention window for a multi-stage pipeline is incorrect.

Under the example [Updating the retention window for a multi-stage pipeline based on stage success](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/run-retention?view=azure-devops&tabs=powershell#example-updating-the-retention-window-for-a-multi-stage-pipeline-based-on-stage-success) the URL and body for the _update_ request is incorrect.

Attempting to use the example currently results in the error:
```
##[error]Invoke-RestMethod : {"count":1,"value":{"Message":"The requested resource does not support http method 'PATCH'."}}
```

The method to update leases is currently only on API version 7.1 Preview 2, and includes moving the lease ID in to the path from the body:

> `PATCH https://dev.azure.com/{organization}/{project}/_apis/build/retention/leases/{leaseId}?api-version=7.1-preview.2`

Additionally the `Update` request does not take an array of objects in the body, but a single object, with the values of `daysValid` and `protectPipeline`.

The example script should therefore be updated to:
```
$rawRequest = @{ daysValid = 365 };
$request = ConvertTo-Json $rawRequest;
$uri = "$(System.CollectionUri)$(System.TeamProject)/_apis/build/retention/leases/$(newLeaseId)?api-version=7.1-preview.2";
```

Which will successfully update the retention lease on a build.

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 46a129c0-8a1c-40c3-47e6-8d95fbce5f26
* Version Independent ID: 8437e895-7164-8585-3b0d-adacd05e6851
* Content: [Using Tasks to Retain Runs in Azure Pipelines - Azure Pipelines](https://learn.microsoft.com/en-us/azure/devops/pipelines/build/run-retention?view=azure-devops&tabs=powershell)
* Content Source: [docs/pipelines/build/run-retention.md](https://github.com/MicrosoftDocs/azure-devops-docs/blob/main/docs/pipelines/build/run-retention.md)
* Service: **azure-devops-pipelines**
* Sub-service: **azure-devops-pipelines-build**
* GitHub Login: @moswald
* Microsoft Alias: **maoswald**

--- a/docs/pipelines/build/run-retention.md
+++ b/docs/pipelines/build/run-retention.md
@@ -156,7 +156,7 @@ To [_update_ a retention lease](/rest/api/azure/devops/build/leases/update) requ
           $contentType = "application/json";
           $headers = @{ Authorization = 'Bearer $(System.AccessToken)' };
           $rawRequest = @{ daysValid = 365 };
-          $request = ConvertTo-Json @($rawRequest);
+          $request = ConvertTo-Json $rawRequest;
           $uri = "$(System.CollectionUri)$(System.TeamProject)/_apis/build/retention/leases/$newLeaseId?api-version=7.1-preview.2";
           Invoke-RestMethod -uri $uri -method PATCH -Headers $headers -ContentType $contentType -Body $request;
 ```

--- a/docs/pipelines/build/run-retention.md
+++ b/docs/pipelines/build/run-retention.md
@@ -155,9 +155,9 @@ To [_update_ a retention lease](/rest/api/azure/devops/build/leases/update) requ
         script: |
           $contentType = "application/json";
           $headers = @{ Authorization = 'Bearer $(System.AccessToken)' };
-          $rawRequest = @{ daysValid = 365; leaseId = $newLeaseId; ownerId = 'User:$(Build.RequestedForId)' };
+          $rawRequest = @{ daysValid = 365 };
           $request = ConvertTo-Json @($rawRequest);
-          $uri = "$(System.CollectionUri)$(System.TeamProject)/_apis/build/retention/leases?api-version=6.0-preview.2";
+          $uri = "$(System.CollectionUri)$(System.TeamProject)/_apis/build/retention/leases/$newLeaseId?api-version=7.1-preview.2";
           Invoke-RestMethod -uri $uri -method PATCH -Headers $headers -ContentType $contentType -Body $request;
 ```
 

